packages/block-editor/src/components/block-mobile-toolbar/index.js

```diff
@@ -8,6 +8,19 @@ import { ifViewportMatches } from '@wordpress/viewport';
  */
 import BlockMover from '../block-mover';
 
+// This is a temporary fix for a couple of issues specific to Webkit on iOS.
+// Without this fix the browser scrolls the mobile toolbar off-screen.
+// Once supported in Safari we can replace this in favor of preventScroll.
+// For details see issue #18632 and PR #18686
+const isIphone = window.navigator.userAgent.indexOf( 'iPhone' ) !== -1;
+if ( isIphone ) {
+	document.addEventListener( 'focusin', function( ) {
+		setTimeout( () => {
+			window.scrollTo( 0, 0 );
+		}, 150 );
+	} );
+}
```

Comment:
While I don't understand the hack entirely, I do feel like maybe it's not the right place for it and maybe something for the edit-post package instead (like in the index or a hack imported there)

Reply:
I'll try to explain the hack to the best of my ability. We might even want to include this as a comment in context of the hack:

- On mobile,`edit-post-editor-regions__body` scrolls to enable a fixed top toolbar.
- On an iOS device, when the mobile keyboard pops up, Mobile Safari forces `html` to scroll upwards, even when it shouldn't, moving the top toolbar out of view.
- This compensates for that onfocus scrolling.